src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -0,0 +1,196 @@
+import EarnFeedItem from 'src/transactions/feed/EarnFeedItem'
+import NftFeedItem from 'src/transactions/feed/NftFeedItem'
+import SwapFeedItem from 'src/transactions/feed/SwapFeedItem'
+import TokenApprovalFeedItem from 'src/transactions/feed/TokenApprovalFeedItem'
+import TransferFeedItem from 'src/transactions/feed/TransferFeedItem'
+import { pendingStandbyTransactionsSelector } from 'src/transactions/reducer'
+import { NetworkId, type TokenTransaction } from 'src/transactions/types'
+import { groupFeedItemsInSections } from 'src/transactions/utils'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+type PaginatedData = {
+  [timestamp: number]: TokenTransaction[]
+}
+
+// Query poll interval
+const POLL_INTERVAL = 10000 // 10 secs
+const FIRST_PAGE_TIMESTAMP = 0
+
+function getAllowedNetworkIdsForTransfers() {
+  return getMultichainFeatures().showTransfers.join(',').split(',') as NetworkId[]
+}
+
+function renderItem({ item: tx }: { item: TokenTransaction; index: number }) {
+  switch (tx.__typename) {
+    case 'TokenExchangeV3':
+    case 'CrossChainTokenExchange':
+      return <SwapFeedItem key={tx.transactionHash} transaction={tx} />
+    case 'TokenTransferV3':
+      return <TransferFeedItem key={tx.transactionHash} transfer={tx} />
+    case 'NftTransferV3':
+      return <NftFeedItem key={tx.transactionHash} transaction={tx} />
+    case 'TokenApproval':
+      return <TokenApprovalFeedItem key={tx.transactionHash} transaction={tx} />
+    case 'EarnDeposit':
+    case 'EarnSwapDeposit':
+    case 'EarnWithdraw':
+    case 'EarnClaimReward':
+      return <EarnFeedItem key={tx.transactionHash} transaction={tx} />
+  }
+}
+
+export default function TransactionFeedV2() {
+  const address = useSelector(walletAddressSelector)
+  const pendingStandByTransactions = useSelector(pendingStandbyTransactionsSelector)
+  const [endCursor, setEndCursor] = useState(0)
+  const [paginatedData, setPaginatedData] = useState<PaginatedData>({})
+  const { pageData, isFetching, error } = useTransactionFeedV2Query(
+    { address: address!, endCursor },
+    {
+      skip: !address,
+      refetchOnMountOrArgChange: true,
+      selectFromResult: (result) => {
+        return {
+          ...result,
+          // eslint-disable-next-line react-hooks/rules-of-hooks
+          pageData: useMemo(
+            () => ({
+              currentCursor: result.originalArgs?.endCursor, // timestamp from the last transaction from the previous page.
+              nextCursor: result.data?.transactions.at(-1)?.timestamp, // timestamp from the last transaction from the current page
+              transactions: result.data?.transactions || [],
+            }),
+            [result]
+          ),
+        }
+      },
+    }
+  )
+
+  // Poll the first page
+  useTransactionFeedV2Query(
+    { address: address!, endCursor: FIRST_PAGE_TIMESTAMP },
+    { skip: !address, pollingInterval: POLL_INTERVAL }
+  )
```

Comment:
If I understand correctly, this will cause the other hook to return new data?
That's why we're not reading the result from it?

Reply:
@jeanregisser Correct. Multiple usages of the same hook with the same arguments will retrieve the data from single cached data point within the api reducer that library manages itself. 

So we can even call this hook in multiple places to retrieve data and if at any moment that data updates _anywhere_ – all the usages will instantly get updated with the new data.
The same goes for this polling example. It's task is only to trigger the fetch every 10 sec and once the data is updated - the hook above with the `selectFromResult` will trigger all the necessary changes and re-renders.